### PR TITLE
Disable "FSE" blocks in Widgets Editor

### LINKED ISSUE #32756: Blocks that use entities are not saved in the widget editors
## Description
When using a full site editing block (e.g. Site tagline) or reusable blocks in the widget editors, changes to those blocks are not being saved.

I imagine the issue is that the widget editor doesn't save all the edited entities. This also exposes the question of whether the editor now needs to replicate the UI that the post and site editors have for entities. I think the answer to that is 'yes', but it might be a challenge in the customizer.

An additional thing to mention is that in the customizer changes to those entity based blocks are not being reflected in the preview

## Step-by-step reproduction instructions
1. Add a Site Tagline or Reusable block
2. Make changes to it
3. Save
4. Reload

## Expected behaviour
Changes to the blocks are kept

In the customizer, changes are shown in the preview

## Actual behaviour
Changes to the blocks are lost

In the customizer, changes are not shown in the preview

### PR DESCRIPTION
**Pull request by getdave**
## Description

Currently FSE blocks don't play well with the Widgets Editor because the required entities aren't there. Moreover, we'd need to support the multi-saving flows/UI from the Site Editor in order to support these blocks.

For 5.8 we could consider disabling all "FSE" blocks. This PR does just that.

It doesn't mean we go with this approach, but this PR allows it to be an option.

Fixes https://github.com/WordPress/gutenberg/issues/32756

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by getdave** [comment]
Can anyone help me to find all the entity based blocks?

**Comment by getdave** [comment]
Now working on disabling in Customizer widgets.

**Comment by Mamaduka** [comment]
The "Site Logo" block doesn't work as well. Changes aren't saved from the widgets screens.

**Comment by talldan** [comment]
I think reusable blocks would also need to be disabled (as they use entities), so I'm not sure how valid this option is.

**Comment by draganescu** [comment]
Yes, ➕ 1️⃣  for reusable block kind of making this option _less_ viable. 

Nevertheless, we _can_ come back later to all entity blocks. Disabling reusable blocks from widget areas now does not prevent them to be enabled later.

**Comment by getdave** [comment]
@draganescu I've removed all the entity blocks including reusable blocks. I'll leave this PR here as an "option" if a better solution can't be found prior to 5.8.

**Comment by noisysocks** [comment]
I don't think we can feasibly implement an entity saving flow in time for 5.8. I'm OK with adding support for these blocks in 5.8.x or 5.9.

**Comment by getdave** [comment]
> I don't think we can feasibly implement an entity saving flow in time for 5.8. I'm OK with adding support for these blocks in 5.8.x or 5.9.

Does that mean we want to look to merge this? If so please could I have a review 🙏 🙇‍♂️

**Review by draganescu — APPROVED** [review]
I tested this and it works as described. Given that this is definitely going to be added back later the code looks good as well. With the exception of reusable blocks, all the theme blocks may still need real world exposure, while widget areas contain all kinds of data today, therefore I believe we're good moving forward with this PR for 5.8. 👍🏻

**Comment by noisysocks** [comment]
There are some conflicts preventing merge. Can you also please open an issue to track re-adding this functionality?

**Comment by getdave** [comment]
Tests failing...again. Unrelated.

See https://github.com/WordPress/gutenberg/pull/32887

**Comment by adamziel** [comment]
Rebased, let's see if the tests pass

**Comment by getdave** [comment]
👀 Thanks for rebase @adamziel. I'll merge ~shortly~ once the React Native tests get fixed in `trunk`. Probably tomorrow.

**Comment by getdave** [comment]
Rebasin' again...

**Comment by getdave** [comment]
Follow up Issue here

https://github.com/WordPress/gutenberg/issues/32952

File: packages/edit-widgets/src/index.js [hunk 1]
```diff
@@ -24,6 +24,16 @@ import './store';
...
 import Layout from './components/layout';
+import {
+	ALLOW_REUSABLE_BLOCKS,
+	ENABLE_EXPERIMENTAL_FSE_BLOCKS,
+} from './constants';
+
+const disabledBlocks = [
+	'core/more',
+	'core/freeform',
+	...( ! ALLOW_REUSABLE_BLOCKS && [ 'core/block' ] ),
```
Comment (draganescu) [inline packages/edit-widgets/src/index.js:35]: @getdave this does not work if we set ALLOW_REUSABLE_BLOCKS to true as the expression will return `false` which is not iterable.

Comment (getdave) [reply packages/edit-widgets/src/index.js:35]: Good spot. Fix available for review https://github.com/WordPress/gutenberg/pull/34634

**Pull request merged**